presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentList.java

```diff
@@ -0,0 +1,89 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.product.launcher.cli;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Module;
+import io.airlift.airline.Command;
+import io.prestosql.tests.product.launcher.Extensions;
+import io.prestosql.tests.product.launcher.LauncherModule;
+import io.prestosql.tests.product.launcher.env.EnvironmentFactory;
+import io.prestosql.tests.product.launcher.env.EnvironmentModule;
+import io.prestosql.tests.product.launcher.env.EnvironmentOptions;
+
+import javax.inject.Inject;
+
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+
+import static io.prestosql.tests.product.launcher.cli.Commands.runCommand;
+import static java.util.Objects.requireNonNull;
+
+@Command(name = "list", description = "lists environments")
+public final class EnvironmentList
+        implements Runnable
+{
+    private final Module additionalEnvironments;
+
+    @Inject
+    public EnvironmentOptions environmentOptions = new EnvironmentOptions();
+
+    public EnvironmentList(Extensions extensions)
+    {
+        this.additionalEnvironments = requireNonNull(extensions, "extensions is null").getAdditionalEnvironments();
+    }
+
+    @Override
+    public void run()
+    {
+        runCommand(
+                ImmutableList.<Module>builder()
+                        .add(new LauncherModule())
+                        .add(new EnvironmentModule(additionalEnvironments))
+                        .add(environmentOptions.toModule())
+                        .build(),
+                EnvironmentList.Execution.class);
+    }
+
+    public static class Execution
+            implements Runnable
+    {
+        private final PrintStream out;
+        private final EnvironmentFactory factory;
+
+        @Inject
+        public Execution(EnvironmentFactory factory)
+        {
+            this.factory = requireNonNull(factory, "factory is null");
+
+            try {
+                this.out = new PrintStream(new FileOutputStream(FileDescriptor.out), true, Charset.defaultCharset().name());
+            }
+            catch (UnsupportedEncodingException e) {
+                throw new IllegalStateException("Could not create print stream", e);
+            }
+        }
+
+        @Override
+        public void run()
+        {
+            out.println("Available environments: ");
+
+            this.factory.list().forEach(out::println);
```

Comment:
no need for `this.` here (and eg  you don't use it for  `out`)

```suggestion
            factory.list().forEach(out::println);
```